### PR TITLE
annotators are here

### PR DESCRIPTION
**Pull request by smautner**
cycle and rna annotation can now be done :)

### CODE REVIEW

File: eden/modifier/graph/cycle.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
+import eden
+from collections import defaultdict
+
+
+class Annotator:
+    def fit(self):
+        pass
```
Comment (bgruening) [inline eden/modifier/graph/cycle.py]: want to raise an https://docs.python.org/3/library/exceptions.html?highlight=notimplementederror#NotImplementedError error?

File: eden/modifier/graph/cycle.py [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+                    return cycle
+                return no_cycle_default
+
+        # we know that the current frontier didntclose cycles so we dont need to look at them again
+        visited = visited | frontier
+        frontier = next
+    return no_cycle_default
+
+
+'''
```
Comment (bgruening) [inline eden/modifier/graph/cycle.py]: put this into a unit-test function ...

Comment (smautner) [inline eden/modifier/graph/cycle.py]: test functions added

**Comment by bgruening** [comment]
Travis is unhappy:

/opt/EDeN/eden/modifier/rna/structure_annotation.py:121:37: E701 multiple statements on one line (colon)

/opt/EDeN/eden/modifier/graph/cycle.py:33:30: W291 trailing whitespace

/opt/EDeN/eden/modifier/graph/cycle.py:36:9: E303 too many blank lines (2)

/opt/EDeN/eden/modifier/graph/cycle.py:53:21: E711 comparison to None should be 'if cond is None:'

**Pull request merged**